pkg/compare/correlator.go

```diff
@@ -228,6 +229,9 @@ func (c *GroupCorrelator) Match(object *unstructured.Unstructured) (*ReferenceTe
 		case len(templates) == 1:
 			return templates[0], nil
 		case len(templates) > 1 && multipleMatchError == nil:
+			if *c.getBestMatch != nil {
```

Comment:
Would it be better to allow `getBestMatch` to return an error and have then none just always return that error?

Would allow times where is undecidable to e.g. when they both have the same number of diffs to also return multiple-match.

Reply:
Infact putting it here means that you'll be doing (N+1) diffs. Where as if you just return multiple values you can just handle the logic at the higher level in the call stack and just keep the best value. 

Perhaps adding a function the to metrics gathering to add multiple match or something like that. 

Another way to get rid of the extra diff is to create a match object which can have an optional diff populated if it is you don't bother doing the diff again.

Reply:
removed the extra diff by changing the metrics gathering